File: 1364-tuple-with-same-product/tuple-with-same-product.java
class Solution {
    public int tupleSameProduct(int[] nums) {
        Map<Integer, Integer> hmap = new HashMap<>();
        int n = nums.length, res = 0;
        for (int i=0; i<n-1; i++) {
            for (int j=i+1; j<n; j++) {
                int prod = nums[i]*nums[j];
                hmap.put(prod, hmap.getOrDefault(prod, 0) + 1);
            }
        }
        for (int cnt: hmap.values()) {
            res += 4*(cnt)*(cnt-1);
        }
        return res;
    }
}

File: 1364-tuple-with-same-product/tuple-with-same-product.py
class Solution:
    def tupleSameProduct(self, nums: List[int]) -> int:
        hmap = defaultdict(int)
        n, res = len(nums), 0
        for i in range(n-1):
            for j in range(i+1, n):
                hmap[nums[i]*nums[j]] += 1
        for cnt in hmap.values():
            res += comb(cnt,2) * 8
        return res